### PR TITLE
Run tests under panic=abort

### PR DESCRIPTION
**Pull request by Darksonn**

### CODE REVIEW

File: ci/panic-abort.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+set -ex
+RUSTFLAGS="$RUSTFLAGS -Cpanic=abort -Zpanic-abort-tests" cargo test --all-features --test test_bytes_vec_alloc -- --nocapture
```
Comment (taiki-e) [inline ci/panic-abort.sh:4]: > --test test_bytes_vec_alloc

Why only this test?

File: tests/test_bytes.rs [hunk 2]
```diff
@@ -291,6 +291,7 @@ fn split_to_uninitialized() {
 }
 
 #[test]
+#[cfg_attr(not(panic = "unwind"), ignore)]
```
Comment (taiki-e) [inline tests/test_bytes.rs:294]:
```suggestion
#[cfg_attr(panic = "abort", ignore)]
```

File: tests/test_bytes.rs [hunk 3]
```diff
@@ -1618,6 +1619,7 @@ fn owned_to_vec() {
...
```
Comment (taiki-e) [inline tests/test_bytes.rs:1622]:
```suggestion
#[cfg_attr(panic = "abort", ignore)]
```

File: ci/panic-abort.sh (hunk 1 above)
Comment (Darksonn) [reply ci/panic-abort.sh]: Debugging artifact. Fixed.

File: tests/test_bytes.rs (hunk 2 above)
Comment (Darksonn) [reply tests/test_bytes.rs:294]: There could be additional panic strategies in the future.

**Pull request merged**